### ISSUE TITLE
The opengl wrapper should produce a dummy dependency to an extension so that loadExtensions always is available

### ISSUE THREAD

**Issue by zacharycarter** [nim-lang/opengl#39]
If loadExtensions is called without any extensions in use, a linker error will be thrown - 

Undefined symbols for architecture x86_64:
  "_nimLoadProcs0", referenced from:
      _Initialize_p4Bl2g9a2WZC1CbGe9b0W2iQ_2 in derelict_game.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

Code to reproduce - 
https://gist.github.com/zacharycarter/24ecddb489b99dd9a45928887d82340b#file-game-nim-L88

Notce on - https://gist.github.com/zacharycarter/24ecddb489b99dd9a45928887d82340b#file-game-nim-L110 that glEnable(GL_DEPTH_TEST) is commented out.

**Comment by krux02** [comment]
well, this is a duplicate of: https://github.com/nim-lang/opengl/issues/26

**Comment by Yepoleb** [comment]
I don't think this is a duplicate. #26 talks about some fallback mechanism to proc loading (which I don't understand) while this issue is about fixing a linker error when no opengl proc is used.

I came across this issue because I was following an OpenGL tutorial where the first step is context creation, which includes initializing OpenGL without any other GL-specific calls. Compilation failed because loadExtensions was the only call from this module. I continued on because it somehow "fixed itself", but I only figured out after months what the real issue was. I think this should be fixed for all the nim newbies like me who try to follow C tutorials and get stuck right at the beginning. Also this applies not just to people who need extensions, loadExtensions is required before any call (bad naming btw).

What this issue does have in common with #26 is that nobody apart from @fowlmouth seems to understand the loader mechanism. Maybe he can explain how it works so this part of the code is not stuck in an unmaintainable state forever.
